cmake/configs/base.cmake

```diff
@@ -400,24 +398,44 @@ endif()
 
 # Setup debug info if enabled.
 if(ENABLE_DEBUG_INFO)
-    if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
-        # Produce full symbolic debugging information.
-        add_compile_options(/Z7)
-        # Ensure a PDB file can be generated for debugging symbols.
-        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DEBUG")
-    else()
+    if(GNU_C_COMPILER OR CLANG_C_COMPILER)
         # Higher debug levels `-g3`/`-ggdb3` emit additional debug information, including
         # macro definitions that allow us to evaluate macros such as `p S2C(session)` inside of gdb.
         # This needs to be in DWARF version 2 format or later - and should be by default - but
         # we'll specify version 4 here to be safe.
-        add_compile_options(-g3)
-        add_compile_options(-ggdb3)
-        add_compile_options(-gdwarf-4)
-        if("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
+        add_cmake_compiler_flags(
+            FLAGS -g3 -gdwarf-4
+            LANGUAGES C CXX
+            BUILD_TYPES Debug RelWithDebInfo
+        )
+        if("${CMAKE_C_COMPILER_ID}" MATCHES "^(Apple)?(C|c?)lang")
             # Clang requires one additional flag to output macro debug information.
-            add_compile_options(-fdebug-macro)
+            add_cmake_compiler_flags(
+                FLAGS -glldb -fdebug-macro
+                LANGUAGES C CXX
+                BUILD_TYPES Debug RelWithDebInfo
+            )
+        else()
+            add_cmake_compiler_flags(
+                FLAGS -ggdb3
+                LANGUAGES C CXX
+                BUILD_TYPES Debug RelWithDebInfo
+            )
         endif()
     endif()
+
+    # MSVC: ensure linker produces PDBs.
+    if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
```

Comment:
Does this line can be replaced with `if(MSVC_CXX_COMPILER)`, similar as what's been done above in CMakeLists.txt?